### PR TITLE
Initiate pip install via PyPI

### PR DESCRIPTION
**Pull request by profxj**
As titled.

See:  https://pypi.org/project/pypeit/

Includes docs for Developers to enable them to 
upload new versions.

### CODE REVIEW

**Comment by profxj** [comment]
@kbwestfall -- Please test for me.  Thanks

**Comment by profxj** [comment]
Address Issue #414

**Comment by kbwestfall** [comment]
Excellent!  I checked in a virtual environment and the pip install works, but it doesn't install all of the dependencies.  I imagine that was on purpose, but I wonder if there's a way we can have pip (or setup.py) check the requirements and report what packages need to be installed or updated.

File: CHANGES.rst [hunk 1]
```diff
@@ -6,6 +6,7 @@
   detected.
 - Revamped the naming convention of output files to have the original
   filename in it.
+- Enable PyPI
```
Comment (kbwestfall) [inline CHANGES.rst]: This is not an edit under this tag of the code.  We should bump the version of master, then pull the change into develop.  We don't necessarily need to retag.

Comment (profxj) [reply CHANGES.rst]: Sorry, I don't follow.  Please dumb it down for me.  Thanks

**Comment by kbwestfall** [comment]
All I meant is to edit the changes and run `bumpversion patch --verbose --commit` .

This moves `master` off of the pip version, but that's okay.  If this works, we may not need the `develop` branch anymore because pip will always point to the most recent version.

**Comment by kbwestfall** [comment]
I just added a script that will checkout the most recent tag of the code.  So if people don't pip install, we should encourage an installation that does:

```
git clone https://github.com/pypeit/PypeIt
cd PypeIt
./checkout_current_tag
python setup.py install
```

**Comment by profxj** [comment]
merging
